plugins/techdocs/src/search/components/TechDocsSearch.tsx

```diff
@@ -72,6 +76,12 @@ const TechDocsSearchBar = (props: TechDocsSearchProps) => {
   } = useSearch();
   const classes = useStyles();
   const [options, setOptions] = useState<any[]>([]);
+
+  const entityTitle = useAsync(async () => {
```

Comment:
Oooh, can we hold off on this until #10178 is in?  That PR introduces a context for entity metadata that can be picked up in here (rather than potentially having to re-load this metadata over the wire).